integrations/SD_Lon/sd_mox.py

```diff
@@ -264,8 +264,10 @@ def _validate_unit_code(self, unit_code, unit_level=None, can_exist=False):
 
         if not code_errors:
             # customers expect unique unit_codes globally
+            # raises if there are more than one on same unit code
+            # one is also bad - this is handled here
             department = self.read_department(unit_code=unit_code)
-            if department is not None and not can_exist:
+            if department is not None:
                 code_errors.append('Enhedsnummer er i brug')
         return code_errors
```

Comment:
Ikke relateret til den aktuelle ændring, men jeg synes flowet i den her metode er lidt forvirrende.

På linje `254` tilføjes en fejlkode til `code_errors` og denne returneres på linje `270`. Pga. flowet i metoden kan der ikke tilføjes mere til `code_errors` på linjerne `255` --> `270`. Jeg ville derfor ændre linje `254` til at returnere fejlen direkte, i stedet for at tilføje til `code_errors` arrayet:
```
return ['Enhedsnummer ikke angivet']
```
Dermed ville jeg også fjerne `else` casen, og droppe de `4` `if/else`'s ud i hovedscopet.

Reply:
Ligeledes ville jeg nok overveje at lave et early return baseret på `code_errors` i stedet for at ekskludere baseret på `code_errors`, altså lave:
```
if not code_errors:
    # Special case
return code_errors
```
til
```
if code_errors:
    return code_errors

department = self.read_department(unit_code=unit_code)
if department is not None:
    return ['Enhedsnummer er i brug']
return []
```

Reply:
På den måde bliver flowet i metoden, tjek for fejl og hvis du finder fejl returnér dem. Hvis du ikke finder fejl, så kør videre indtil du når bunden af metoden, hvor der ingen fejl var.